content/blogposts/2022/sourcegraph-4.0.md

```diff
@@ -0,0 +1,163 @@
+---
+title: "Sourcegraph 4.0"
+publishDate: 
+description: Announcing Sourcegraph 4.0
+tags: [blog, release]
+slug: "release/4.0"
+published: false
+heroImage: 
+socialImage: 
+changelogItems:
+---
+
+# Sourcegraph 4.0
+
+> NOTE: This post is very WIP. It's a rough draft of what individual teams intend to ship.
+
+- [From code search toward a code intelligence platform](/blog/code-search-to-code-intelligence) - the longer-term vision/story
+- [Secure, easy, and scalable Sourcegraph Cloud managed instances for the enterprise](/blog/enterprise-cloud) - the Cloud announcement (including SOC2 Type 2, etc.) that coincides with 4.0
+
+## TL;DR
+
+`TODO(sqs): The section headers with emojis are what I think we should use, but the wording of the bullet points within are WIP`
+
+##### 🌅 Code intelligence: more useful, more times daily for every dev
+
+- The fastest and most scalable code search (search perf, page load perf, scalability: # repos, big monorepos, # code hosts, etc.)
+- TODO how to introduce the simple UI?
+- Smarter code search
+- High-level aggregations of search results
+- Precise code navigation for 9 languages (adding Ruby and Rust) with auto-indexing
+
+##### 🏗️ High-leverage ways to improve your entire codebase
+
+- Run batch changes server side
+- Something about the new packaging with batch changes and code insights
+
+##### ☁️ Enterprise Cloud dedicated(/managed?) instances
+
+- [Secure, easy, and scalable Sourcegraph Cloud dedicated(/managed?) instances for the enterprise](/blog/enterprise-cloud)
+
+##### 💖 For admins
+
+- Admin analytics
+- 1-click log export
+- OpenTelemetry
```

Comment:
> Easily export traces using OpenTelemetry

Reply:
Easily export traces using OpenTelemetry

[OpenTelemetry (OTEL)](https://opentelemetry.io/) is an open specification for exporting observability data into a user’s preferred systems. We aim to make OpenTelmetry the default way to quickly and easily export observability data from Sourcgraph, starting with tracing in Sourcegraph 4.0. This is a breaking change since we now require customers to export tracing using the [OpenTelemetry collector](https://opentelemetry.io/docs/collector/), which will now be deployed by default in all Sourcegraph deployment methods, replacing the Jaeger agents and collectors we used to ship by default.